book/src/csi-noide-info-object.md

```diff
@@ -0,0 +1,59 @@
+# CSI CRD: `CSINodeInfo` Object
+
+## What is the `CSINodeInfo` object?
+
+CSI drivers generate node specific information. Instead of storing this in the Kubernetes `Node` API Object, a new CSI specfici Kubenretes CRD was create, the `CSINodeInfo` CRD.
+
+It serves the following purposes:
+
+1. Mapping Kubernetes node name to CSI Node name,
+  * The CSI `GetNodeInfo` call returns the name by which the storage system refers to a node. Kubernetes must use this name in future `ControllerPublishVolume` calls. Therefore, when a new CSI driver is registered, Kubernetes stores the storage system node ID in the `CSINodeInfo` object for future reference. 
+2. Driver availability
+  * A way for kubelet to communicate to the kube-controller-manager and kubernetes scheduler whether the driver is available (registered) on the node or not.
+3. Volume topology
+  * The CSI `GetNodeInfo` call returns a set of key/value labels identifying the topology of that node. Kubernetes uses this information to to do topology aware provisioning, so it stores the key/values as labels on the Kubernetes node object. In order to recall which `Node` label keys belong to a specific CSI driver, the kubelet stores the keys in the `CSINodeInfo` object for future reference.
+	
+
+## What fields does the `CSINodeInfo` object have?
+
+Here is an example of a v1alpha1 `CSINodeInfo` object:
+
+```YAML
+apiVersion: csi.storage.k8s.io/v1alpha1
+kind: CSINodeInfo
+metadata:
+  name: node1
+spec:
+  drivers:
+  - name: mycsidriver.example.com
+    available: true
+    volumePluginMechanism: csi-plugin
+status:
+  drivers:
+  - name: mycsidriver.example.com
+    nodeID: storageNodeID1
+    topologyKeys: ['mycsidriver.example.com/regions', "mycsidriver.example.com/zones"]
+```
+
+Where the fields mean:
+
+- `csiDrivers` - list of CSI drivers running on the node and their properties.
+- `driver` - the CSI driver that this object refers to.
+- `nodeID` - the assigned identifier for the node as determined by the driver.
+- `topologyKeys` - A list of topology keys assigned to the node as supported by the driver.
+
+## What creates the `CSINodeInfo` object?
+
+CSI drivers do not need to create the `CSINodeInfo` object directly. As long as they use the [node-driver-registrar](node-driver-registrar.md) sidecar container, the kubelet will automatically populate the `CSINodeInfo` object for the CSI driver as part of kubelet plugin registration.
+
+### Enabling `CSINodeInfo`
+
+The `CSINodeInfo` object is available as alpha starting with Kubernetes v1.12. Because it is an alpha feature, it is disabled by default.
```

Comment:
Should we put at the top of this page the alpha/beta/ga status?

Reply:
Good idea.